srcts/src/components/sidebar.ts

```diff
@@ -297,24 +326,61 @@ class Sidebar {
     });
   }
 
+  /**
+   * Retrieves the current window size by reading a CSS variable whose value is
+   * toggled via media queries.
+   * @returns The window size as `"desktop"` or `"mobile"`, or `""` if not
+   * available.
+   */
+  private _getWindowSize(): SidebarWindowSize | "" {
+    const { container } = this.layout;
+
+    return window
+      .getComputedStyle(container)
+      .getPropertyValue("--bslib-sidebar-js-window-size")
+      .trim() as SidebarWindowSize | "";
+  }
+
   /**
    * Initialize the sidebar's initial state when `open = "desktop"`.
    * @private
    */
-  private _initDesktop(): void {
+  private _initSidebarState(): void {
     const { container } = this.layout;
-    // If sidebar is marked open='desktop'...
-    if (container.dataset.bslibSidebarOpen?.trim() !== "desktop") {
+
+    const initDesktop = container.dataset.openDesktop?.trim();
+    const initMobile = container.dataset.openMobile?.trim();
+
+    // If initial states are identical, the markup is correct
+    if (initDesktop === initMobile) {
       return;
     }
 
-    // then close sidebar on mobile
-    const initCollapsed = window
-      .getComputedStyle(container)
-      .getPropertyValue("--bslib-sidebar-js-init-collapsed");
+    // Check the CSS variable to find out which mode we're in right now
+    this.windowSize = this._getWindowSize();
+
+    const initAttr = this.windowSize === "desktop" ? initDesktop : initMobile;
+    if (!initAttr) {
+      this.toggle("open", true);
+      return;
+    }
```

Comment:
If we need to handle `undefined`, let's make that more clear (and maybe even just throw an error?)